### DISCUSSION TITLE
Does compiling my commandline app to an executable make my code private?

### DISCUSSION THREAD

**Discussion by jrohlandt** [denoland/deno#14871] [Q&A]
Does Deno compile down to binary, bytecode or something?  E.g. if I run "deno compile myfile.ts". Will my logic inside of myfile.exe be unreadable/un-extractable by malicious users? Or can my code be unpacked to expose the .js files?

I'm thinking about writing my business logic as a commandline application that I will bundle with my Electron app. The main reason for this is that I want to keep my source code (business logic) private (or at least make it very difficult to copy and tinker with).

But pretty big secondary benefit is that I can reuse the commandline app if I ever need to ditch Electron.

**Comment by AnInternetTroll** [comment]
No, it's practically plain text. You can even open the compiled binary in vim and go at the end and you'll see your code with comments and everything

**Reply by jrohlandt** [reply]
Thanks. I tested it by opening the .exe in Notepad and I see my JS code at the end like you said. Cheers!

**Reply by NetOpWibby** [reply]
Dang, would be neat if it did.
